pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -271,9 +227,11 @@ func (n *clusterManagerController) sync(ctx context.Context, controllerContext f
 
 	// If there are some invalid feature gates of registration, will output
 	// condition `ValidRegistrationFeatureGates` False in ClusterManager.
-	if clusterManager.Spec.RegistrationConfiguration != nil {
-		featureGates, condition, err := n.checkFeatureGate(ctx, clusterManagerName,
-			clusterManager.Spec.RegistrationConfiguration.FeatureGates, "registration")
+	if clusterManager.Spec.RegistrationConfiguration != nil &&
+		len(clusterManager.Spec.RegistrationConfiguration.FeatureGates) > 0 {
```

Comment:
why we need to check empty? empty is a valid setting.

Reply:
when it is empty, I  think we do not need to set the condition for `ValidRegistrationFeatureGates` or `ValidWorkFeatureGates`

Reply:
I think condition should always be set. And if user set feature is empty. We need to change the deployment config. It is different from nil.